### DISCUSSION TITLE
Twitter introducing caps of 10k tweets/requests for $100 / month and deprecating v1.1 API

### DISCUSSION THREAD

**Discussion by timminata** [plhery/node-twitter-api-v2#459] [General]
https://twitter.com/TwitterDev/status/1641222782594990080

Major blow to all those who rely on the Twitter API. From ~2m tweets to ~10k and with a massive $100 fee.
Any plans to update this package to remove v1.1 support, or any other changes? Seems the rate limits will also be a major pain to deal with.

**Comment by alkihis** [comment]
Indeed this is a major change in the Twitter API ecosystem.
v1.1 support in the near future is mostly unknown yet, even in new Entreprise tier(s).

Regarding the extremely hostile attitude of newly acquired Elon Musk's Twitter against third party developers, I (main developer of this library) don't want to support it anymore, and therefore, any future update (including deprecations and documentation updates) is uncertain.
Nobody will pay an absurd amount of money (100$) to get only a v2 access (that still miss 50% of v1.1 features) for 0.5% of the previous tweet polling rates (10K vs 2M for "Elevated" free tier).

API has not been touched (in a programmatically sense of thinking) since the massive layoffs of Twitter employees in november 2022 and, regarding the changes already been made in the main "offical" apps, that are only botch job and rebranding of other features, I personally don't think the API can evolve in the right direction.

So, if the only possible remaining users are: 
- v2-API-only users that can only fetch their own profile details once an hour (see "Free" tier) and posts tweets without ability to read any replies or interactions
- entreprises that can afford the thousands of dollars "Entreprise" tiers (see the glorious Google Forms available on developer.twitter.com in order to apply to one of them!) thus they can also literaly *pay* developers to maintain their own Twitter-communication-API-system or whatever

and if Twitter cannot maintain their own roadmap to migrate all features of v1.1 API to the v2 before dropping it, as they planned before the takeover, what's the point?

With all my regret, I don't want to be a part of this adventure anymore.

**Reply by barca-reddit** [reply]
Just wanted to say massive thanks for developing and maintaining this in first place. I remember how helpful you were with one of my issues, you went great lengths just to try and help me and I really appreciate that.

Regarding Twitter, you have to be out of your mind to pay them these absurd asking prices each month and on top of that develop a library for them for free, so your decision is completely expected and understandable.

Anyway, I just wanted to say I wish you all the best!

**Reply by saturdaywalkers** [reply]
@alkihis And a thank you from me too!. 
Just had to change libs to this one as a result of the API changes. Aside: async/await is **so** much simpler to use! ).
Lost half the functionality (geo searches are "elevated" and no longer free).
May I suggest you add an example of tweeting an image (i.e. v1 media upload, then v2 tweet).

**Reply by Zedifuu** [reply]
Following on from Barca as well, @alkihis I remember as well you taking the time to help me troubleshoot stream reconnection issues when I started using the library, You have done magnificent work maintaining this library. 

It is a real shame but I think this may be the end of my project, like I can afford the $100 a month basic but is it really worth it, for my little discord bot that crossposts to a few streamer discords? Probably not.... I don't really know if I'd want to feed money the guy destroying the once thriving open source twitter development community.

Elon is an absolute trumpet.

**Reply by Zedifuu** [reply]
Actually no.. filtered stream which is what I use for the crossposting is only in pro... 5000/month.. definitely not.

**Comment by kylekz** [comment]
Is 1.1 completely dead now? I migrated to the "new" free tier a while ago but now v1 media uploads are returning 403/453..

**Reply by timminata** [reply]
Yes, 1.1 is dead now. I'm still using the v2 endpoints. You get 1.5K free fetches a month, and then 10k for $100/month. It's not really viable for hobbyists though - but better than nothing I guess 😬 Also, the premium/elevated tiers have all been removed (at least on the account I had)

**Reply by kylekz** [reply]
My use case is well within limits, the issue is that the only way to post media was to use the v1.1 media/create endpoint to get a `media_id`, then use that in a v2 tweet create call. Now v1.1 is dead, there's no way to post media via v2 that I can see.

Edit: Actually seems I was mistaken. v1.1 still works for media uploads, I was still using v1.1 for posting the actual tweet. Switching to v2 for the tweet creation works.

**Reply by ceosss** [reply]
@kylewardnz ,
Did you use oauth1 or oauth2?

**Comment by doniyoraloxanov** [comment]
Hello guys I need help !

I'm working on a Telegram bot that generates images, and I want users to be able to share these images on their Twitter accounts. I've successfully uploaded the image to Twitter and obtained the media_id, but I'm struggling to figure out how to attach this image to a tweet using Twitter Web Intents.

I've referred to the Twitter Web Intents documentation (https://dev.twitter.com/docs/intents), but I couldn't find a clear example or explanation on how to include media in the tweet using Web Intents.

Could someone provide guidance or a code example on how to use Twitter Web Intents to attach an image to a tweet programmatically?
